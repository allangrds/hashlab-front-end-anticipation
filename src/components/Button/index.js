import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.module.css'

const Button = ({
  disabled,
  onClick,
  text,
}) => (
  <button
    className={styles.button}
    type="button"
    onClick={onClick}
    disabled={disabled}
  >
    { text }
  </button>
)

Button.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
}

Button.defaultProps = {
  disabled: false,
}

export default Button
