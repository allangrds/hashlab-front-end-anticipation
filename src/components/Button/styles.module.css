@import "../../assets/css/index.css";

.button {
  background-color: var(--color-blue-1);
  border: none;
  border-radius: 4px;
  cursor: pointer;
  font-family: "Source Sans Pro", sans-serif;
  font-size: 16px;
  color: var(--color-white);
  padding: 10px 14px;
  transition: 0.1s;
}

.button:hover {
  background-color: var(--color-blue-2);
}
