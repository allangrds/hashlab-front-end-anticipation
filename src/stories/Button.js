import React from 'react'
import PropTypes from 'prop-types'
import './button.css'

const Button = ({
  backgroundColor, label, primary, size, ...props
}) => {
  const mode = primary ? 'storybook-button--primary' : 'storybook-button--secondary'
  return (
    <button
      type="button"
      className={['storybook-button', `storybook-button--${size}`, mode].join(' ')}
      style={backgroundColor && { backgroundColor }}
      {...props}
    >
      {label}
    </button>
  )
}

export default Button

Button.propTypes = {
  backgroundColor: PropTypes.string,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
}

Button.defaultProps = {
  backgroundColor: null,
  onClick: undefined,
  primary: false,
  size: 'medium',
}
