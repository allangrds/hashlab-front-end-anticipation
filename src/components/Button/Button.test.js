import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Button from '.'

test('Button should render and be updated correctly', () => {
  const callback = jest.fn()
  const text = 'Bot√£o'

  const {
    container,
    getByRole,
  } = render(
    <Button
      text={text}
      onClick={callback}
    />,
  )

  expect(callback).toHaveBeenCalledTimes(0)

  const button = getByRole('button')
  userEvent.click(button)

  expect(callback).toHaveBeenCalledTimes(1)

  expect(container).toMatchSnapshot()
})
