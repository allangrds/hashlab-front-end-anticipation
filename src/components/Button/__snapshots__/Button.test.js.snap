// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and be updated correctly 1`] = `
<div>
  <button
    class="button"
    type="submit"
  >
    Bot√£o
  </button>
</div>
`;
